react/hooks/useFacetNavigation.js

```diff
@@ -0,0 +1,68 @@
+import { useCallback, useContext } from 'react'
+import { useRuntime } from 'vtex.render-runtime'
+
+import QueryContext from '../components/QueryContext'
+import { HEADER_SCROLL_OFFSET } from '../constants/SearchHelpers'
+
+const scrollOptions = {
+  baseElementId: 'search-result-anchor',
+  top: -HEADER_SCROLL_OFFSET,
+}
+
+const removeElementAtIndex = (str, index, separator) =>
+  str
+    .split(separator)
+    .filter((_, i) => i !== index)
+    .join(separator)
+
+const useFacetNavigation = () => {
+  const { navigate } = useRuntime()
+  const { query, map } = useContext(QueryContext)
+
+  const navigateToFacet = useCallback(
+    maybeFacets => {
+      const facets = Array.isArray(maybeFacets) ? maybeFacets : [maybeFacets]
+
+      const { currentQuery, currentMap } = facets.reduce(
+        ({ currentQuery, currentMap }, facet) => {
+          if (facet.selected) {
+            const facetIndex = currentQuery
+              .toLowerCase()
+              .split('/')
+              .map(decodeURIComponent)
+              .findIndex(
+                value => value === decodeURIComponent(facet.value).toLowerCase()
+              )
+
+            return {
+              currentQuery: removeElementAtIndex(currentQuery, facetIndex, '/'),
+              currentMap: removeElementAtIndex(currentMap, facetIndex, ','),
+            }
+          } else {
```

Comment:
useless else

Reply:
I think it reads better this way but oh well